.github/workflows/main.yaml

```diff
@@ -0,0 +1,32 @@
+name: Automated testing
+
+on:
+  push:
+    branches:
+      - "*"
+      - "!test*"
+jobs:
+  run-checks:
+    runs-on: ubuntu-latest
+    outputs:
+      cache_key: ${{ steps.texlive.outputs.cache_key }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - run: sudo apt-get update && sudo apt-get install ghostscript
+      - name: Install TeX Live
+        id: texlive
+        uses: zauguin/install-texlive@v3
+        with:
+          # List the required TeX Live packages in a separate file to allow reuse in
+          # different workflows.
+          package_file: .github/tl_packages
+      - run: l3build install
```

Comment:
for all the comments about the action workflow: that was a quick copy and paste to have some external test runs at all and should be reviewed (also in tagpdf and pdfresources and hyperref) generally. This can be done later after a release.